app/models/concerns/host_searching.rb

```diff
@@ -98,10 +98,7 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
     end
 
     def filter_by_compliance(_filter, operator, value)
-      ids = ::Host.includes(test_results: :profile).select do |host|
-        host.compliant.values.all?(::ActiveModel::Type::Boolean.new.cast(value))
-      end
-      ids = ids.pluck(:id).map { |id| "'#{id}'" }
+      ids = compliant_host_ids(value).pluck(:host_id).map { |id| "'#{id}'" }
```

Comment:
We can drop pluck and map and use a subselect to speed it up even further.

Reply:
How can we do that? Considering what I wrote below :thinking:

Reply:
```
operator = operator == '<>' ? 'NOT' : ''
{ conditions: "hosts.id #{operator} IN (#{compliant_host_ids(value).to_sql})" }
```
would that work?

Reply:
Umm, we're not building SQL query here, the conditions are in the query language of the scoped search. I don't really want to stretch it too far :see_no_evil:

Reply:
See other searches in this same file. The 'query language' used by scoped search is essentially ActiveRecord. Anything put in `conditions` here will be added to the `where` clause. Using a sub-query is definitely preferred than listing all the IDs because the `IN (...)` SQL syntax has a maximum length that we could run up against if the user has many compliance hosts.